### PR TITLE
TASK: Refactor ImagineFactory

### PR DESCRIPTION
**Pull request by kitsunet**
Allows to check for driver existence and instantiation of any
imagine class.

### CODE REVIEW

File: Classes/ImagineFactory.php [hunk 1]
```diff
@@ -37,41 +38,86 @@ class ImagineFactory extends AbstractImagineFactory
...
-            case 1: $object = new $className($arguments[0]); break;
-            case 2: $object = new $className($arguments[0], $arguments[1]); break;
-            case 3: $object = new $className($arguments[0], $arguments[1], $arguments[2]); break;
-            case 4: $object = new $className($arguments[0], $arguments[1], $arguments[2], $arguments[3]); break;
-            case 5: $object = new $className($arguments[0], $arguments[1], $arguments[2], $arguments[3], $arguments[4]); break;
-            case 6: $object = new $className($arguments[0], $arguments[1], $arguments[2], $arguments[3], $arguments[4], $arguments[5]); break;
-            default:
-                $class = new \ReflectionClass($className);
-                $object =  $class->newInstanceArgs($arguments);
-        }
```
Comment (mhsdesign) [inline Classes/ImagineFactory.php:62]: :eyes:

good to have that gone

**Review by mhsdesign — APPROVED** [review]
By reading ;)

File: Classes/ImagineFactory.php (hunk 1 above)
Comment (kitsunet) [reply Classes/ImagineFactory.php:62]: welcome to how our object manager looked a few years back 😂 It was faster though because the reflection is slow.

**Review by daniellienert — APPROVED** [review]
Looks good by reading

**Pull request merged**